### ISSUE TITLE
Add Languages Email Template

### ISSUE THREAD

**Issue by jujes** [reactioncommerce/reaction#2256]
#### Enhancements
-----

> The only way to see messages from the site is in english? as default template?
> Some possibility to add other languages templates and send it using the language what the visitor has choosen for navigate??

**Thanks in advance,**

**Comment by zenweasel** [comment]
We should probably add this to the docs that the templates need to be added, that the underlying templates collection supports languages. @kieckhafer Anything special someone needs to do to add more language-based templates?

**Comment by kieckhafer** [comment]
You'll need to register the template with the corresponding language in `/imports/plugins/included/email-templates/server/index.js` to make sure it gets dumped into the `Templates` collection.

Looks like we also need to make some adjustments to choose which language gets sent, shouldn't be more than just adding a couple variables to the `SSR.complieTemplate` function that runs [here](https://github.com/reactioncommerce/reaction/blob/master/server/methods/core/orders.js#L574).

**Comment by xelfe** [comment]
So what we have to use when we register a email template in other language than english ?

Is it the right way to use the language tag into registerTemplate ?

 Reaction.registerTemplate({
  title: "Accounts - Reset Password",
  name: TemplatePaths.resetPaswordTemplate,
  type: "email",
  template: Reaction.Email.getTemplateFile(TemplatePaths.resetPaswordTemplate),
  language: "English",
  subject: "{{shop.name}}: Here's your password reset link"
});

**Comment by kieckhafer** [comment]
You'd use the [ISO language code](https://www.w3schools.com/tags/ref_language_codes.asp), instead of the full name:

```
Reaction.registerTemplate({
title: "Accounts - Reset Password",
name: TemplatePaths.resetPaswordTemplate,
type: "email",
template: Reaction.Email.getTemplateFile(TemplatePaths.resetPaswordTemplate),
language: "en",
subject: "{{shop.name}}: Here's your password reset link"
});
```

To change the subject, you could change it here if you'd like, or you can also change it in the Dashboard once the template is registered.

**Comment by xelfe** [comment]
thank you @kieckhafer. What about adding template into custom path ? Is it possible for the email or I should override files into private/email/templates/ ?

**Comment by xelfe** [comment]
@kieckhafer wrote;

> You'd use the ISO language code, instead of the full name:
> 
```
Reaction.registerTemplate({
title: "Accounts - Reset Password",
name: TemplatePaths.resetPaswordTemplate,
type: "email",
template: Reaction.Email.getTemplateFile(TemplatePaths.resetPaswordTemplate),
language: "en",
subject: "{{shop.name}}: Here's your password reset link"
});
```
> To change the subject, you could change it here if you'd like, or you can also change it in the Dashboard once the template is registered.

I've tried and into the dashboard, the language is still stuck to english... I've overrided the default e-mail templates into import/plugins/custom/email-template. See example below;

```Reaction.registerTemplate({
  title: "Default",
  name: TemplatePaths.coreDefaultTemplate,
  type: "email",
  template: Reaction.Email.getTemplateFile(TemplatePaths.coreDefaultTemplate),
  language: "fr",
  subject: "Un message de {{shop.name}}"
});
```

**Comment by jujes** [comment]
Hi @zenweasel @kieckhafer, 

I've tried and into the dashboard, the language continue on **en**

![language_template_issue](https://user-images.githubusercontent.com/2881017/30853604-8fa5dabe-a2af-11e7-880c-fb74b193ed48.png)

**imports/plugins/included/email-templates/lib/paths.js**
`export const esResetPaswordTemplate = "accounts/esResetPassword";`
  
**imports/plugins/included/email-templates/server/index.js**
```
  Reaction.registerTemplate({
    title: "Recuperar mi contraseña",
    name: TemplatePaths.esResetPaswordTemplate,
    type: "email",
    template: Reaction.Email.getTemplateFile(TemplatePaths.esResetPaswordTemplate),
    language: "es",
    subject: "{{shop.name}}: Aquí tienes el enlace para reestablecer tu cotraseña"
    });
```
**and create the .html file**
private/email/templates/accounts/esResetPassword.html

**Comment by kieckhafer** [comment]
Is the language showing up in the database?

**Comment by joelquatro** [comment]
for anyone interested... setting language won't happen to make it to the db as the function processTemplateInfoForDatabase(templateInfo) is not returning that field.

The trace would be:
Reaction.registerTemplate -> registerTemplateForDatabase -> processTemplateInfoForDatabase

code in server/api/core/templates.js, line 160:
export function processTemplateInfoForDatabase(templateInfo) {
  const templateData = {
    name: templateInfo.name,
    title: templateInfo.title,
    type: templateInfo.type,
    subject: templateInfo.subject,
    templateFor: templateInfo.templateFor
  };

  if (typeof templateInfo.template === "string") {
    templateData.template = templateInfo.template;
    templateData.parser = TEMPLATE_PARSER_HANDLEBARS;
  } else if (typeof templateInfo.template === "object") {
    templateData.template = templateInfo.template;
    templateData.parser = TEMPLATE_PARSER_REACT;
  } else if (typeof templateInfo.template === "function") {
    templateData.parser = TEMPLATE_PARSER_REACT;
  }

  return templateData;
}

**Comment by aldeed** [comment]
I believe this is fixed and/or should work properly after #5171 is merged. If not, we can reopen
